### PR TITLE
Newly added contacts will be separated from imported contacts #573

### PR DESCRIPTION
**Pull request by florianhaar**
**Link to the given issue**
Fixes https://github.com/open-xchange/ox-coi/issues/573

**Describe what the problem was / what the new feature is**
A contact, which is automatically added to contacts, will be added separated from imported contacts.

**Describe your solution**
Changed the result of the requested contacts. Now a `List<dynamic>` is returned from `ContactListStateSuccess`. This list contains a `String` which is the header for the group or a `ValueKey` which contains the id and last update value for the contacts itself. The returned list in `ContactListStateSuccess` contains the new list which is called `contactElements`. 

**Additional context**
- There is only one Bloc to handle all contact item requests. The `ContactChangeBloc` and the `ContactImportBloc` were merged into the `ContactItemBloc`
- The contact group header is now added by the Lists itself and not by each `ContactListItem`

### CODE REVIEW

File: lib/src/contact/contact_item_event_state.dart [hunk 1]
```diff
@@ -41,50 +41,222 @@
...
+class DeleteContact extends ContactItemEvent {
+  final int id;
+
+  DeleteContact({@required this.id});
+
+  @override
+  List<Object> get props => [id];
+}
+
+class ContactAdded extends ContactItemEvent {
```
Comment (Boehrsi) [inline lib/src/contact/contact_item_event_state.dart:87]: No needed anymore, please remove.

File: lib/src/contact/contact_item_event_state.dart [hunk 2]
```diff
@@ -41,50 +41,222 @@
...
+  final int id;
+
+  ContactAdded({@required this.id});
+
+  @override
+  List<Object> get props => [id];
+}
+
+class ContactEdited extends ContactItemEvent {
```
Comment (Boehrsi) [inline lib/src/contact/contact_item_event_state.dart:96]: No needed anymore, please remove.

File: lib/src/contact/contact_item_event_state.dart [hunk 3]
```diff
@@ -41,50 +41,222 @@
...
+  @override
+  List<Object> get props => null;
+}
 
-  ContactLoaded({@required this.name, @required this.email, @required this.color, @required this.isVerified, @required this.imagePath, this.phoneNumbers, this.headerText});
+class ContactDeleted extends ContactItemEvent {
```
Comment (Boehrsi) [inline lib/src/contact/contact_item_event_state.dart:101]: No needed anymore, please remove.

File: lib/src/contact/contact_item_event_state.dart [hunk 4]
```diff
@@ -41,50 +41,222 @@
...
+  final int chatId;
+  final int messageId;
+
+  BlockContact({this.chatId, this.id, this.messageId});
+
+  @override
+  List<Object> get props => [id, chatId, messageId];
+}
+
+class ContactBlocked extends ContactItemEvent {
```
Comment (Boehrsi) [inline lib/src/contact/contact_item_event_state.dart:128]: No needed anymore, please remove.

File: lib/src/contact/contact_item_event_state.dart [hunk 5]
```diff
@@ -41,50 +41,222 @@
...
+class UnblockContact extends ContactItemEvent {
+  final int id;
+
+  UnblockContact({@required this.id});
+
+  @override
+  List<Object> get props => [id];
+}
+
+class ContactUnblocked extends ContactItemEvent {
```
Comment (Boehrsi) [inline lib/src/contact/contact_item_event_state.dart:146]: No needed anymore, please remove.

File: lib/src/contact/contact_list_content.dart [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    this.contactItemType,
+    this.selectedContacts,
+    this.callback,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final key = getKeyFromContactElement(contactElement);
+    final idOrHeader = extractId(key);
+    final isInt = idOrHeader is int;
```
[the remark this answers is not in the document]
Comment (florianhaar) [reply lib/src/contact/contact_list_content.dart:84]: Sounds good

**Pull request merged**